src/app/Library/CrudPanel/Traits/Relationships.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace Backpack\CRUD\app\Library\CrudPanel\Traits;
+
+use Exception;
+use Illuminate\Database\Eloquent\Relations\Relation;
+
+trait Relationships
+{
+    /**
+     * If the field name is not a relationship method e.g: article_id,
+     * we try to find if this field has a relation defined.
+     *
+     * @param string $fieldName
+     * @return array|bool
+     */
+    public function checkIfFieldNameBelongsToAnyRelation($fieldName)
+    {
+        $relations = $this->getAvailableRelationsInModel();
+
+        if (empty($relations)) {
+            return false;
+        }
+
+        if (in_array($fieldName, array_column($relations, 'name'))) {
+            return array_filter($relations, function ($arr) use ($fieldName) {
+                if (isset($arr['name'])) {
+                    return $arr['name'] == $fieldName;
+                }
+
+                return false;
+            })[0];
+        }
+
+        return false;
+    }
+
+    /**
+     * Get the user defined methods in model that return any type of relation.
+     *
+     * @return array
+     */
+    public function getAvailableRelationsInModel()
```

Comment:
@pxpm are you sure this works? In my tests it does not. And I don't think it's me who broke it 😬 Not this time at least.

I've tested this on the Article model, which has two relationships. But it always returns an empty array. Because of this, ```checkIfFieldNameBelongsToAnyRelation()``` above also fails.

Is this supposed to work, or was it just a work-in-progress?

Reply:
For this to work you need to provide return types in your relation definition.

```php
public function article() : BelongsTo {
        return $this->belongsTo('Backpack\NewsCRUD\app\Models\Article');
    }
```

Reply:
OOOH! Of course. Now I understand 😄 

I don't think this is something we can count on, though:
- the [Laravel docs on Relationships](https://laravel.com/docs/6.x/eloquent-relationships) do not instruct people to use return types;
- the generators we currently use don't have return types;

Since it's not something we can count on, I'm not sure it's worth including... What do you think?